### ISSUE TITLE
Update OpenBLAS version to the latest one

### ISSUE THREAD

**Issue by marcinz** [nv-legate/cunumeric#26]
OpenBLAS released version 0.3.15. We should update to the latest version.

**Comment by piyueh** [comment]
I having an issue using the default flags in `install.py` to build OpebBLAS v0.3.15. One of OpenBLAS's unit tests failed: the one in [`utest/test_post_fork.c`](https://github.com/xianyi/OpenBLAS/blob/904f9a267dddb30e9f187e57231ed160ab2f2704/utest/test_post_fork.c). The failed unit test blocks the installation of OpenBLAS.

For me, the problem is caused by the hard-coded flag `NUM_PARALLEL=32` in legate.core's `install.py`. If I change it to match the number of logical cores on may machine, which is just 12, then the unit test works, and then installation can be finished.

Actually, the same unit test in v0.3.13 also failed. However, for some reason I don't know, the unit test was never launched when installing v0.3.13. And that's why I didn't have an installation issue with v0.3.13.

**Comment by lightsighter** [comment]
@piyueh I think this is a new test that OpenBLAS has added recently. I believe that this is actually an issue that should be reported to OpenBLAS. `NUM_PARALLEL` just says how many concurrent calls can come into OpenBLAS at the same time:
https://github.com/xianyi/OpenBLAS/blob/develop/Makefile.rule#L93-L98
There's no reason that that should be limited by the number of physical cores. It would be perfectly reasonable for a client to have more threads than cores and make many concurrent calls into OpenBLAS.

**Comment by piyueh** [comment]
I think that unit test existed in v0.3.13 but was not triggered on my machine. All unit tests in the folder `OpenBLAS/utest` are only launched when variable `${CROSS}` not equal to `1` ([line 51 in OpenBLAS/utest/Makefile](https://github.com/xianyi/OpenBLAS/blob/d2b11c47774b9216660e76e2fc67e87079f26fa1/utest/Makefile#L51)). Not sure why, when I installed v0.3.13, `${CROSS}`  is `1`, so the tests were not triggered. But when I installed v0.3.15, `${CROSS}` became undefined, so the tests were launched and failed.

I'll submit an issue to OpenBLAS  once I have more insight to the issue.
